File: PHP/DesignPattern/FactoryMethod/ChicagoPizzaStore.php
<?php

	include 'PizzaStore.php';
	include 'ChicagoStyleCheesePizza.php';
	include 'ChicagoStyleVegglePizza.php';

	/**
	* 具体创建者类
	* 继承自工厂方法类, 能产生产品的类
	*/
	class ChicagoPizzaStore extends PizzaStore {

		/**
		* 实现抽象方法createPizza，产生具体产品
		*/
		public function createPizza($type) {
			if ($type == "cheese") {
				return new ChicagoStyleCheesePizza();
			} else if ($type == "veggle") {
				return new ChicagoStyleVegglePizza();
			} else {
				return null;
			}
		}
	}

File: Java/FactoryMethod/src/com/base/ChicagoPizzaStore.java
package com.base;

/**
 * Created by hhq on 1/14/15.
 */
public class ChicagoPizzaStore extends PizzaStore {
    protected Pizza createPizza(String item) {
        if (item.equals("cheese")) {
            return new ChicagoStyleCheesePizza();
        } else if (item.equals("veggie")) {
            return new ChicagoStyleCheesePizza();
        } else if (item.equals("clam")) {
            return new ChicagoStyleClamPizza();
        } else if (item.equals("pepperoni")) {
            return new ChicagoStylePepperoni();
        } else
            return null;
    }
}
